Research and Data Pack Update (#588)
ldtteam/MinecoloniesWiki: source/misc/datapacks.md

---

> title: Data Packs Customization

Reviewer: ```suggestion
title: Data Packs
```

layout: default
---

> # Data Pack Customization

Reviewer: ```suggestion
# Data Packs
```

- [Concepts](#concepts)
  - [Terminology](#terminology)
  - [Example Folder Layout](#example-folder-layout)
- [Tags](#tags)
  - [Block Tags](#block-tags)
  - [Item Tags](#item-tags)
  - [Crafter Item Tags](#crafter-item-tags)
- [Crafter Recipes](#crafter-recipes)
- [Player Recipes](#player-recipes)
- [Loot Tables](#loot-tables)
- [Research Customization](#research-customization)
  - [Branches](#branches)
  - [Research](#research)
  - [Research Effects](#research-effects)
  - [MineColonies Research Effects](#minecolonies-research-effects)
  - [MineColonies Building Unlocks](#minecolonies-building-unlocks)

> MineColonies allows modifications of many features using data packs, including player and worker recipes, loot tables, and mob drops. This allows broad expansion by players or modpack makers to support other mods, design choices, forms of progression, or styles of play. For general information on MineCraft Data Packs, [see the official wiki](<https://minecraft.fandom.com/wiki/Data_Pack>).

Reviewer: ```suggestion
MineColonies allows modifications of many features using data packs, including player and worker recipes, loot tables, and mob drops. This allows broad expansion by players or modpack makers to support other mods, design choices, forms of progression, or styles of play. For general information on Minecraft data packs, see the [official wiki](https://minecraft.fandom.com/wiki/Data_Pack).
```

> Data Packs exist as part of a world, and they must be [installed](https://minecraft.fandom.com/wiki/Tutorials/Installing_a_data_pack) on each world.

Reviewer: ```suggestion
Data packs exist as part of a world, and they must be [installed](https://minecraft.fandom.com/wiki/Tutorials/Installing_a_data_pack) on each world.
```

## Concepts

> Data Packs are one or more files in the JSON format, stored within a folder or a zip file. Despite their names, these are text files, and can be opened with text editors. Note that Windows will, by default, hide the extension of known files, and this [should be changed](https://support.microsoft.com/en-us/windows/common-file-name-extensions-in-windows-da4a4430-8e76-89c5-59f7-1cdbbc75cb01) to avoid accidentally appending .txt to the file name. Avoid using a rich text editor like WordPad or Microsoft Word, which may insert additional formatting into the file. On Windows, simple text editors like Notepad, NotePad++, or WattPad are more useful for making small numbers of these files, and development environments like IntelliJ may be worth installing if creating many JSONs.

Reviewer: ```suggestion
Data packs are one or more files in the JSON format, stored within a folder or a zip file. Despite their names, these are text files, and can be opened with text editors. Note that Windows will, by default, hide the extension of known files, and this [should be changed](https://support.microsoft.com/en-us/windows/common-file-name-extensions-in-windows-da4a4430-8e76-89c5-59f7-1cdbbc75cb01) to avoid accidentally appending .txt to the file name. Avoid using a rich text editor like WordPad or Microsoft Word, which may insert additional formatting into the file. On Windows, simple text editors like Notepad, NotePad++, or WattPad are more useful for making small numbers of these files, and development environments like IntelliJ may be worth installing if creating many JSONs.
```

> The JSON format is both generous and fastiduous. It does not particularly care if you add extra fields, but it will choke on a missing comma or brace. MineCraft will report the file and location of a JSON error in the error log, but it may also be useful to check files in a JSON validator tool, colloqually known as a [linter](https://jsonlint.com/), as you create them. The JSON format will accept most characters, though the double-quotes (<code>"</code>) and backslash (<code>\</code>) characters must first be 'escaped' by prefixing them with a backslash (such that <code>"</code> becomes <code>\"</code> and <code>\\</code> becomes <code>\\</code>).

Reviewer: ```suggestion
The JSON format is both generous and fastidious. It does not particularly care if you add extra fields, but it will choke on a missing comma or brace. Minecraft will report the file and location of a JSON error in the error log, but it may also be useful to check files in a JSON validator tool, colloqually known as a [linter](https://jsonlint.com/) as you create them. The JSON format will accept most characters, though the double-quotes (<code>"</code>) and backslash (<code>\</code>) characters must first be "escaped" by prefixing them with a backslash (such that <code>"</code> becomes <code>\"</code> and <code>\\</code> becomes <code>\\</code>).
```

> For most users, looking at other similar JSONs will be the fastest way to get started. For those interested in the specific rules of the format, [see here](<https://www.json.org/json-en.html>).

Reviewer: ```suggestion
For most users, looking at other similar JSONs will be the fastest way to get started. For those interested in the specific rules of the format, [see here](https://www.json.org/json-en.html).
```

> The Data Pack folder or zip file can be any valid file name, and will be used to determine the name of the Data Pack. This folder or zip file must contain in its root level a file named **pack.mcmeta**, [with a specific format](<https://minecraft.gamepedia.com/Data_Pack#pack.mcmeta>). It is strongly encouraged to provide a distinct name and description for your Data Pack: this will show up as a tooltip from the in-game interfaces and /datapack list command. To act as a Data Pack, this should also contain a "data" directory.

Reviewer: ```suggestion
The Data pack folder or zip file can be any valid file name and will be used to determine the name of the Data Pack. This folder or zip file must contain in its root level a file named **pack.mcmeta** [with a specific format](https://minecraft.gamepedia.com/Data_Pack#pack.mcmeta). It is strongly encouraged to provide a distinct name and description for your data pack: this will show up as a tooltip from the in-game interfaces and /datapack list command. To act as a data pack, this should also contain a "data" directory.
```
